matrix_content_scanner/scanner/scanner.py

```diff
@@ -47,6 +52,25 @@ def __init__(self, mcs: "MatrixContentScanner"):
             strict=True
         )
 
+        # Result cache settings.
+        cache_ttl = parse_user_value(
+            mcs.config.result_cache.ttl,
+            parser=parse_timespan,
+        )
+        # We know cache_ttl can't be None because the relevant configuration setting can't
+        # be None.
+        assert cache_ttl is not None
+
+        self._result_cache: TTLCache[str, CacheEntry] = TTLCache(
```

Comment:
we should probably have some docstrings on `CacheEntry` saying what the fields mean and when they are populated.